psi4/driver/frac.py

```diff
@@ -352,67 +418,92 @@ def frac_nuke(molecule, **kwargs):
+    core.print_out("""    %6s %6s %6s %6s %6s %6s\n""" % ('N', 'Na', 'Nb', 'Charge', 'Mult', 'HOMO'))
     for line in stats:
         core.print_out(line)
 
-    core.print_out('\n\t"You shoot a nuke down a bug hole, you got a lot of dead bugs"\n')
-    core.print_out('\t\t\t-Starship Troopers\n')
+    core.print_out('\n    "You shoot a nuke down a bug hole, you got a lot of dead bugs"\n')
+    core.print_out('            -Starship Troopers\n')
 
     # Drop the files out
-    fh = open(traverse_filename, 'w')
-    fh.write("""\t%-11s %-24s %-24s %11s\n""" % ('N', 'Energy', 'HOMO Energy', 'Converged'))
-    for k in range(len(Ns)):
-        fh.write("""\t%11.3E %24.16E %24.16E %11d\n""" % (Ns[k], energies[k], potentials[k], convs[k]))
-    fh.close()
+    with open(traverse_filename, 'w') as fh:
+        fh.write("""    %-11s %-24s %-24s %11s\n""" % ('N', 'Energy', 'HOMO Energy', 'Converged'))
+        for k in range(len(Ns)):
+            fh.write("""    %11.3E %24.16E %24.16E %11d\n""" % (Ns[k], energies[k], potentials[k], convs[k]))
 
-    fh = open(stats_filename, 'w')
-    fh.write("""\t%6s %6s %6s %6s %6s %6s\n""" % ('N', 'Na', 'Nb', 'Charge', 'Mult', 'HOMO'))
-    for line in stats:
-        fh.write(line)
-    fh.close()
+    with open(stats_filename, 'w') as fh:
+        fh.write("""    %6s %6s %6s %6s %6s %6s\n""" % ('N', 'Na', 'Nb', 'Charge', 'Mult', 'HOMO'))
+        for line in stats:
+            fh.write(line)
 
     # Properly, should clone molecule but since not returned and easy to unblemish,
     molecule.set_molecular_charge(charge0)
     molecule.set_multiplicity(mult0)
 
+    optstash.restore()
     return E
 
-def ip_fitting(molecule, omega_l, omega_r, **kwargs):
-    kwargs = p4util.kwargs_lower(kwargs)
 
-    # By default, zero the omega to 3 digits
-    omega_tol = kwargs.get('omega_tolerance', 1.0E-3)
+def ip_fitting(name, omega_l=0.05, omega_r=2.5, omega_convergence=1.0e-3, maxiter=20, **kwargs):
+    """Optimize DFT omega parameter for molecular system.
+
+    Parameters
+    ----------
+    name : string, functional function
+        DFT functional string name or function defining functional
+        whose omega is to be optimized.
+    omega_l : float, optional
+        Minimum omega to be considered during fitting.
+    omega_r : float, optional
+        Maximum omega to be considered during fitting.
+    molecule : :ref:`molecule <op_py_molecule>`, optional
+        Target molecule (neutral) for which omega is to be tuned, if not last defined.
+    omega_convergence : float, optional
+        Threshold below which to consider omega converged. (formerly omega_tolerance)
+    maxiter : int, optional
+        Maximum number of iterations towards omega convergence.
+
+    Returns
+    -------
+    float
+        Optimal omega parameter.
+
+    """
+    optstash = p4util.OptionsState(
+        ['SCF', 'GUESS'],
+        ['SCF', 'DF_INTS_IO'],
+        ['SCF', 'DFT_OMEGA'],
```

Comment:
Good to pop DFT_OMEGA!

Reply:
During the options overhaul it could be nice to have a signature like `set_global_option(option, parameter, stash)` which would continuously build a stash so that you wouldn't forget to reset these values. Although I guess if we have the ability to have local options thats a mute point.